Add upgrade guide entry for TaskContainer.create deprecation (#31814)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

To determine the location of the Mac OS X SDK for Swift, Gradle now passes the `--sdk macosx` arguments to `xcrun`.
This is necessary because the SDK could be discovered inconsistently without this argument across different environments.

> ==== Source level deprecation of TaskContainer.create methods

Reviewer: can we link to the javadoc for TaskContainer somewhere?

Eager task creation methods on the `TaskContainer` interface have been marked `@Deprecated` and will generate compiler and IDE warnings when used in build scripts or plugin code.
There is not yet a Gradle deprecation warning emitted for their use.

> However, for builds that have configured to fail on warnings - either Kotlin script or plugin code compilation - this may cause the build to fail.

Reviewer 2: ```suggestion
However, for builds that have been configured to fail on warnings during either Kotlin script or plugin code compilation, this may cause the build to fail.
```

> However, for builds that have configured to fail on warnings - either Kotlin script or plugin code compilation - this may cause the build to fail.

Reviewer 3: ```suggestion
However, if the build is configured to fail on warnings during Kotlin script or plugin code compilation, this behavior may cause the build to fail.
```

> A full Gradle deprecation of these methods will happen in a later version.

Reviewer 2: ```suggestion
A standard Gradle deprecation warning will be printed upon use when these methods are fully deprecated in a later version.
```

> A full Gradle deprecation of these methods will happen in a later version.

Reviewer 3: ```suggestion
A standard Gradle deprecation warning will be printed when these methods are fully deprecated in a future version.
```
